javascripts/content_scripts/posting_button.js

```diff
@@ -1,131 +1,323 @@
 /*global privlyUrlReceiptNode:false, pendingPost:false, chrome:false, ls:true,  */
```

Comment:
I know it wasn't here before, but some sort of header describing the file's purpose and any general implementation notes for the future would be useful. Perhaps also include a list of the types of elements the button should and should not work on.